client/octree_viewer.ts

```diff
@@ -86,78 +95,146 @@ class NodeLoader {
   private threePoints: THREE.Points;
 
   constructor(public nodeName: string) {
-    this.displayedLevel = -1;
-    this.fetchingLevel = -1;
+    this.diplayedLevelOfDetail = -1;
+    this.fetchingLevelOfDetail = -1;
     this.threePoints = undefined;
   }
 
   public isUpToDate(lod: number) {
-    if (this.fetchingLevel === lod) {
+    if (this.fetchingLevelOfDetail === lod) {
       return true;
     }
-    if (this.fetchingLevel === -1 && this.displayedLevel === lod) {
+    if (this.fetchingLevelOfDetail === -1 && this.diplayedLevelOfDetail === lod) {
       return true;
     }
     return false;
   }
 
-  public startFetching(lod: number) { this.fetchingLevel = lod; }
+  public startFetching(lod: number) { this.fetchingLevelOfDetail = lod; }
 
   public newData(
-      scene: THREE.Scene, material: THREE.ShaderMaterial, lod: number,
-      nodePoints: NodePoints) {
+      scene: THREE.Scene, commonMaterial: THREE.ShaderMaterial, lod: number,
+      nodeRenderData: NodeRenderData) {
     // If this node contains no points.
-    if (nodePoints.points.length === 0) {
+    if (nodeRenderData.position.length === 0) {
       return;
     }
 
-    if (lod !== this.fetchingLevel) {
+    if (lod !== this.fetchingLevelOfDetail) {
       return;
     }
-    this.fetchingLevel = -1;
+    this.fetchingLevelOfDetail = -1;
 
-    if (this.displayedLevel === lod) {
+    if (this.diplayedLevelOfDetail === lod) {
       return;
     }
-    this.displayedLevel = lod;
+    this.diplayedLevelOfDetail = lod;
 
     if (this.threePoints !== undefined) {
       scene.remove(this.threePoints);
     }
 
     const geometry = new THREE.BufferGeometry();
-    // itemSize = 3 because there are 3 values (components) per vertex
+    // itemSize = 3 because there are 3 values (components) per vertex.
     geometry.addAttribute(
-        'position', new THREE.BufferAttribute(nodePoints.points, 3));
+        'position', new THREE.BufferAttribute(nodeRenderData.position, 3, nodeRenderData.normalizePosition));
     geometry.addAttribute(
-        'color', new THREE.BufferAttribute(nodePoints.colors, 4));
+        'color', new THREE.BufferAttribute(nodeRenderData.color, 3));
+
+    // THREE can no longer figure out the bounding box or the bounding sphere of
+    // this node, since the 'position' attribute does not contain it. So we
+    // help it out.
+    geometry.boundingBox = new THREE.Box3(
+        nodeRenderData.min,
+        new THREE.Vector3(
+            nodeRenderData.min.x + nodeRenderData.edgeLength,
+            nodeRenderData.min.y + nodeRenderData.edgeLength,
+            nodeRenderData.min.z + nodeRenderData.edgeLength));
+    geometry.boundingSphere = new THREE.Sphere(
+        new THREE.Vector3(
+            nodeRenderData.min.x + nodeRenderData.edgeLength / 2,
+            nodeRenderData.min.y + nodeRenderData.edgeLength / 2,
+            nodeRenderData.min.z + nodeRenderData.edgeLength / 2),
+        nodeRenderData.edgeLength / 2
```

Comment:
This is a sphere contained inside the bounding box, right? And is this okay? Or should it contain the bounding box instead?